shopping_mall/backend/app/services/faq_sync.py

```
@@ -0,0 +1,201 @@
+        except Exception as e:
+            logger.error(
+                "[faq_sync] delete 실패: id=%s collection=%s error=%s",
+                chroma_doc_id, chroma_collection, e,
+            )
+
+    # ── BM25 인덱스 재빌드 ──────────────────────────────────────────────────
+
+    @staticmethod
+    def _rebuild_bm25_debounced(client=None) -> None:
+        """30초 디바운스 — 대량 upsert 시 BM25를 매번 재빌드하지 않음."""
+        global _last_bm25_rebuild
```

Ruff (0.15.12)

[warning] Using the global statement to update `_last_bm25_rebuild` is discouraged

(PLW0603)